mmeds/server.py

```diff
@@ -807,8 +814,21 @@ def login(self, username, password):
         try:
             validate_password(username, password, testing=self.testing)
             cp.session['user'] = username
-            cp.session['temp_dir'] = tempfile.TemporaryDirectory()
-            cp.session['working_dir'] = Path(cp.session['temp_dir'].name)
+
+            # TODO: TEmporary for figuring this out
+
+            path = fig.DATABASE_DIR
+            filename = cp.session['user']
+            # Create the filename
+            temp_dir = Path(path) / Path(filename).name
+
+            # Ensure there is not already a file with the same name
+            while temp_dir.is_dir():
+                temp_dir = Path(path) / '_'.join([fig.get_salt(5), Path(filename).name])
```

Comment:
If you're getting a unique path name with the get_salt(5) function, then presumably the while loop will only run once? If that's the case, why have a while loop or am I missing something?

Reply:
99.9% of the time you would expect that to be the case. But a string of length 5, it's only 25^5 possible combinations, very unlikely to repeat but not impossible.

Reply:
ah makes sense